numba/extending.py

```diff
@@ -71,6 +71,10 @@ def len_impl(seq):
 
     Compiler options can be passed as an dictionary using the **jit_options**
     argument.
+
+    Overloading strictness (that the typing and implementing signatures match)
+    is enforced by the **strict** keyword argument, it is recommended that this
+    is set to True (default).
```

Comment:
Would we be deprecating **strict** in the future?  If so, we should mention that in the docstring to make sure code will be refactored to follow the strict rules.

Reply:
I think possibly not?! there's cases where it's actually useful at present https://github.com/numba/numba/pull/3648/files#r247949928, one way around may be to have the public `@overload` not have this option, and it call a private one that has such an option.